### ISSUE TITLE
Request for provider!  Cardinia, Victoria, Australia

### ISSUE THREAD

**Issue by vytux-com** [mampfes/hacs_waste_collection_schedule#413]
Can we please add this to the source list?

https://www.cardinia.vic.gov.au/info/20002/rubbish_and_recycling/385/bin_collection_days_and_putting_your_bins_out/2

**Comment by skipishere** [comment]
Looks like the initial search using "Daly Rd, Nar Nar Goon Vic" as an example hits the page https://geocoder.cit.api.here.com/6.2/search.json?app_id=pYZXmzEqjmR2DG66DRIr&app_code=T-Z-VT6e6I7IXGuqBfF_vQ&country=AUS&state=VIC&&searchtext=Daly%20Rd%2C%20Nar%20Nar%20Goon%20Vic
(the App id and app code seem to be needed, unsure if they change with usage)

```json
{
    "Response": {
        "MetaInfo": {
            "Timestamp": "2022-11-08T23:02:48.131+0000"
        },
        "View": [
            {
                "_type": "SearchResultsViewType",
                "ViewId": 0,
                "Result": [
                    {
                        "Relevance": 0.86,
                        "Distance": 2886.4,
                        "MatchLevel": "street",
                        "MatchQuality": {
                            "Country": 1.0,
                            "State": 1.0,
                            "City": 1.0,
                            "Street": [
                                1.0
                            ]
                        },
                        "Location": {
                            "LocationId": "NT_CwL.5zhRWw6bCkcZtyEzDB",
                            "LocationType": "address",
                            "DisplayPosition": {
                                "Latitude": -38.10658,
                                "Longitude": 145.60191
                            },
                            "NavigationPosition": [
                                {
                                    "Latitude": -38.10658,
                                    "Longitude": 145.60191
                                }
                            ],
                            "MapView": {
                                "TopLeft": {
                                    "Latitude": -38.10543,
                                    "Longitude": 145.58621
                                },
                                "BottomRight": {
                                    "Latitude": -38.11707,
                                    "Longitude": 145.61092
                                }
                            },
                            "Address": {
                                "Label": "Daly Rd, Nar Nar Goon VIC 3812, Australia",
                                "Country": "AUS",
                                "State": "VIC",
                                "City": "Nar Nar Goon",
                                "Street": "Daly Rd",
                                "PostalCode": "3812",
                                "AdditionalData": [
                                    {
                                        "value": "Australia",
                                        "key": "CountryName"
                                    },
                                    {
                                        "value": "Victoria",
                                        "key": "StateName"
                                    }
                                ]
                            }
                        }
                    }
                ]
            }
        ]
    }
}
```

From this I'm unsure but looks like it uses the Location and then either the DisplayPosition or NavigationPosition Latitude & Longitude to then built up a url of
https://services3.arcgis.com/TJxZpUnYIJOvcYwE/arcgis/rest/services/WasteCollectionZones/FeatureServer/0/query?f=geojson&outFields=*&returnGeometry=true&inSR=4326&spatialRel=esriSpatialRelIntersects&geometryType=esriGeometryPoint&geometry=145.60191000000003%2C-38.10658000000001

Which in this example bring back json like this (I've removed a pile of coordinates that are returned in the json for brevity)

```json
{
    "type": "FeatureCollection",
    "features": [
        {
            "type": "Feature",
            "id": 12,
            "geometry": {
                "type": "Polygon",
                "coordinates": [
                    [
                        [
                            145.755351999479,
                            -38.0611520000684
                        ],
                        [
                            145.756611000509,
                            -38.061281999633
                        ],
                        [
                            145.756389999661,
                            -38.0614199998287
                        ],
                        [
                            145.755219000208,
                            -38.0621579996841
                        ],
                       ...
                        [
                            145.755351999479,
                            -38.0611520000684
                        ]
                    ]
                ]
            },
            "properties": {
                "FID": 12,
                "OBJECTID": 12,
                "grn_day": "Tuesday",
                "grn_weeks": 2,
                "grn_start": "2017-08-01",
                "rec_day": "Tuesday",
                "rec_weeks": 2,
                "rec_start": "2017-08-08",
                "rub_day": "Tuesday",
                "rub_weeks": 1,
                "rub_start": "2017-08-01",
                "rec_name": "Week 2",
                "grn_name": "Week 2",
                "Note": " ",
                "Shape_STAr": 242889145.193,
                "Shape_STLe": 111025.33941,
                "Shape__Area": 392430357.546875,
                "Shape__Length": 141161.356638329
            }
        }
    ]
}
```

As I'm unfamiliar with the way their bin scheduling works, is it a case of checking if it's a week 1 or 2 and alternating base on that? The website shows the following at the time of writing this 
![image](https://user-images.githubusercontent.com/26724777/200697397-220023d6-6a70-4482-a37d-1ce20631893f.png)

**Comment by vytux-com** [comment]
Not sure about just alternating as it would pork most of the time except in leap years when there are 53 weeks (2020 being last example) 

I am fairly certain that that year the recycling would have switch from odd to even or visa versa depending on your address. That is on week 53 you would have let's say recycling and the following week 1 would have green waste.

**Comment by skipishere** [comment]
Sorry I should have been clearer about working out week 1 or week 2, as like you say 53 week years would cause issues. I meant to calculate for a given date if (based off the start date in 2017) if it was a week 1 or 2.

Having used the website today, I think I understand the data returned a bit better.
For rubbish 
```json
"rub_day": "Tuesday",
"rub_weeks": 1,
"rub_start": "2017-08-01",
```
I can see it is a weekly thing, and recycling it's every 2 weeks, so maybe it as nice as working out how many weeks since the `-start` date then divide by `_weeks` and then if there is no remainder it is due that week. I'll try and have a play about with the logic tonight and see what I can come up with 😄

**Comment by seedzero** [comment]
https://github.com/mampfes/hacs_waste_collection_schedule/pull/712

**Comment by mampfes** [comment]
fixed by #712
